composer.json

```diff
@@ -15,6 +15,7 @@
     "dealerdirect/phpcodesniffer-composer-installer": "^0.7",
     "phpcompatibility/php-compatibility": "^9.3",
     "phpunit/phpunit": "^4|^5|^6|^7|^8|^9",
+    "slevomat/coding-standard": "^8.0",
```

Comment:
This coding standard requires php 7.2. Will would make this plugin break in 7.2- no. 

https://github.com/slevomat/coding-standard/blob/f69e2524e8770efb9b3e5ac4a0ebc0d54eb446d7/composer.json#L19

Reply:
It would not break the plugin, because the composer autoloader is only used in tests.

It would just break tests running on PHP 5.6. However, right now tests are not running on this version on CI, so... 🤷

Reply:
As I understand, there are three possible ways to about this:

1. Have a separate composer file for PHP 7.2+
2. In GHA workflow, just-in-time `composer require` this package before doing `composer install` before running PHPCS
3. In GHA workflow, just-in-time `composer remove` this package before doing `composer install` prior to running PHPUnit in PHP<7.2.

I think the third option is ideal, since all normal development environments would be running PHP 7.2+. It's only the exception case, specifically on GHA, where PHP 5.6 is needed. This would allow a developer to have all packages installed normally. 

The first option doesn't seem ideal since there would be a separate composer file to maintain, and Dependabot wouldn't automatically be aware of it.

Reply:
How about we address this issue as part of #399?

Reply:
We may not want this sniff after all. See https://github.com/WordPress/wordpress-develop/pull/4457#discussion_r1194463282

Reply:
Noting that we could still make a separate decision from core here, there are several plugins that have stricter requirements than core to enforce better coding practices. I don't feel strongly about this one, but I feel like if we can find a way that we can integrate it that remains compatible with PHP 5.6, I'd be happy to include it.

Reply:
I am not against using this rule set, but php 5.6 compat is the key here.

Reply:
Can the 5.6 compat be handled as part of #399? Without CI actually running anything in 5.6, it makes it a bit difficult to verify the compat is working, or at least it seems a bit premature.